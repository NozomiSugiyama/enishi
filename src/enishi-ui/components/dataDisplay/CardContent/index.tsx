import React from "react";

import { styled } from "styled-components";

import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

const Host = styled(LinearLayout)`
    && {
        padding: 0.5rem 1rem 1rem;
    }
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout>;

export const CardContent: React.FC<Props> = ({ className, children, ...props }) => (
    <Host className={className} orientation="vertical" {...props}>
        {children}
    </Host>
);
