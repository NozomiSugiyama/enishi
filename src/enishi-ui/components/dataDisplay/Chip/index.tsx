import React from "react";

import { CircleChip } from "@/enishi-ui/components/dataDisplay/Chip/CircleChip";
import { NormalChip } from "@/enishi-ui/components/dataDisplay/Chip/NormalChip";

type Props =
    | (React.ComponentPropsWithRef<typeof CircleChip> & { circle: true })
    | (React.ComponentPropsWithRef<typeof NormalChip> & { circle?: false });

export const Chip: React.FC<Props> = (props) => {
    // eslint-disable-next-line react/destructuring-assignment
    if (props.circle) {
        const { circle: _, ...circleChipProps } = props;
        return <CircleChip {...circleChipProps} />;
    }

    const { circle: _, ...defaultChipProps } = props;
    return <NormalChip {...defaultChipProps} />;
};
