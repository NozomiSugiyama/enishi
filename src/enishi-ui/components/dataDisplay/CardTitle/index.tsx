import React from "react";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";

type Props = React.ComponentPropsWithRef<typeof Typography> & {
    readonly children?: React.ReactNode;
};

export const CardTitle: React.FC<Props> = ({ children, ...props }) => (
    <Typography colorTheme="secondary" {...props}>
        {children}
    </Typography>
);
