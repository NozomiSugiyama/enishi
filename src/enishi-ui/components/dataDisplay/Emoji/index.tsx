import React from "react";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";

type Props = React.ComponentPropsWithRef<typeof Typography> & {
    readonly emoji: string;
};

export const Emoji: React.FC<Props> = ({ emoji, children, ...props }) => (
    <Typography {...props}>
        <span role="img">{emoji}</span>
        {children}
    </Typography>
);
