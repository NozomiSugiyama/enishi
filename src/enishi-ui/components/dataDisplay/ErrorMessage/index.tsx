import React from "react";

import { MdError } from "@react-icons/all-files/md/MdError";
import { styled, useTheme } from "styled-components";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";
import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

const Host = styled(LinearLayout)`
    && {
        margin-left: 0.2rem;
        margin-top: 0.1rem;
        > :first-child {
            margin-right: 0.2rem;
        }
    }
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout>;

export const ErrorMessage: React.FC<Props> = ({ children, ...props }) => {
    const theme = useTheme();
    return (
        <Host gravity="center_left" {...props}>
            <MdError color={theme.palette.error[100]} />
            <Typography color={theme.palette.error[100]} data-w-id="form-item-container-error-message-field" variant="caption">
                {children}
            </Typography>
        </Host>
    );
};
