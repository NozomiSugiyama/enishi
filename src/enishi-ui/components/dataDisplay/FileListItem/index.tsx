import React from "react";

import { FiFileText } from "@react-icons/all-files/fi/FiFileText";

import { ListItem } from "@/enishi-ui/components/dataDisplay/ListItem";
import { ListItemIcon } from "@/enishi-ui/components/dataDisplay/ListItemIcon";

export const FileListItem: React.FC<React.ComponentPropsWithRef<typeof ListItem>> = ({ children, ...props }) => (
    <ListItem {...props}>
        <ListItemIcon size={props.size}>
            <FiFileText />
        </ListItemIcon>
        {children}
    </ListItem>
);
