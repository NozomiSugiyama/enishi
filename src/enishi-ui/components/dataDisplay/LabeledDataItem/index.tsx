import React from "react";

import { styled } from "styled-components";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";
import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

const sizes = {
    "x-small": {
        host: {
            padding: "0.3rem 0",
            fontSize: "0.8rem"
        },
        label: {
            marginBottom: "0.1rem",
            fontSize: "0.7rem"
        },
        iconSize: "1rem"
    },
    "small": {
        host: {
            padding: "0.4rem 0",
            fontSize: "0.9rem"
        },
        label: {
            marginBottom: "0.1rem",
            fontSize: "0.8rem"
        },
        iconSize: "1rem"
    },
    "medium": {
        host: {
            padding: "0.5rem 0",
            fontSize: "1rem"
        },
        label: {
            marginBottom: "0.4rem",
            fontSize: "0.8rem"
        },
        iconSize: "1rem"
    },
    "large": {
        host: {
            padding: "0.6rem 0",
            fontSize: "1.2rem"
        },
        label: {
            marginBottom: "0.4rem",
            fontSize: "0.8rem"
        },
        iconSize: "1rem"
    },
    "x-large": {
        host: {
            padding: "0.7rem 0",
            fontSize: "1.6rem"
        },
        label: {
            marginBottom: "0.4rem",
            fontSize: "0.9rem"
        },
        iconSize: "1rem"
    }
};

const Host = styled(LinearLayout)<{
    $size: keyof typeof sizes;
}>`
    && {
        padding: ${({ $size }) => sizes[$size].host.padding};
        font-size: ${({ $size }) => sizes[$size].host.fontSize};
    }
`;

const Header = styled(LinearLayout).attrs({
    orientation: "horizontal"
})<{
    $size: keyof typeof sizes;
}>`
    margin-bottom: ${({ $size }) => sizes[$size].label.marginBottom};
`;

const Label = styled(Typography)<{
    $size: keyof typeof sizes;
}>`
    && {
        white-space: nowrap;
        font-size: ${({ $size }) => sizes[$size].label.fontSize};
    }
`;

const IconContainer = styled.div<{ $size: keyof typeof sizes }>`
    display: flex;
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout> & {
    readonly label?: string;
    readonly labelProps?: React.ComponentPropsWithRef<typeof Typography>;
    readonly size?: keyof typeof sizes;
    readonly fontWeight?: React.ComponentPropsWithRef<typeof Typography>["fontWeight"];
    readonly labelButton?: React.ReactNode | null;
};

export const LabeledDataItem: React.FC<Props> = ({ label, children, labelProps, size = "medium", fontWeight, labelButton, ...props }) => (
    <Host $size={size} orientation="vertical" {...props}>
        <Header $size={size} gravity="center">
            <Label $size={size} colorTheme="secondary" fontWeight={fontWeight} variant="caption" {...labelProps}>
                {label}
            </Label>
            {labelButton ? <IconContainer $size={size}>{labelButton}</IconContainer> : null}
        </Header>
        {children}
    </Host>
);
