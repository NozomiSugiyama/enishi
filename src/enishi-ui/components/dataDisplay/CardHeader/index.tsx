import React from "react";

import { styled } from "styled-components";

import { CardTitle } from "@/enishi-ui/components/dataDisplay/CardTitle";
import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

type Size = "large" | "medium";
const Host = styled(LinearLayout)<{ size: Size }>`
    padding: ${({ size }) => (size === "medium" ? "0.5rem 1rem 0.3rem" : "1rem 1.5rem 0.5rem")};
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout> & {
    readonly size?: Size;
    readonly onClose?: () => void;
};

export const CardHeader: React.FC<Props> = ({ size = "medium", children, ...props }) => (
    <Host gravity="center_left" orientation="horizontal" size={size} {...props}>
        {typeof children === "string" ? <CardTitle variant={size === "large" ? "h4" : "h6"}>{children}</CardTitle> : children}
    </Host>
);
