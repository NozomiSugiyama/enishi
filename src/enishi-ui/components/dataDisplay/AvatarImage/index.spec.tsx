import React from "react";

import { describe, expect, it } from "@jest/globals";
import "jest-styled-components";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { AvatarImage } from "@/enishi-ui/components/dataDisplay/AvatarImage";

import { createThemes } from "@/theme";

const { theme } = createThemes("#1F4682");

describe("<AvatarImage />", () => {
    it("should merge user classes & spread custom props to the root node", () => {
        const component = renderer.create(
            <ThemeProvider theme={theme}>
                <AvatarImage className="my-avatar" data-my-prop="woofAvatar">
                    Avatar
                </AvatarImage>
            </ThemeProvider>
        );
        let tree = component.toJSON() as renderer.ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();

        void renderer.act(() => {
            tree.props as React.ComponentPropsWithRef<typeof AvatarImage>;
        });
        tree = component.toJSON() as renderer.ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
    });
    it("should display a skeleton when loading", () => {
        const component = renderer.create(
            <ThemeProvider theme={theme}>
                <AvatarImage loading>Loading Avatar</AvatarImage>
            </ThemeProvider>
        );
        const tree = component.toJSON() as renderer.ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
    });
    it("should render as a button when button", () => {
        const component = renderer.create(
            <ThemeProvider theme={theme}>
                <AvatarImage button>Button Avatar</AvatarImage>
            </ThemeProvider>
        );
        const tree = component.toJSON() as renderer.ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
    });
});
