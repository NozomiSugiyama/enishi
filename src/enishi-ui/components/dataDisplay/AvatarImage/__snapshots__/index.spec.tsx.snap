// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarImage /> should display a skeleton when loading 1`] = `
.c2 {
  position: absolute;
  box-sizing: border-box;
  background-color: rgba(0, 0, 0, 0.12);
  border-radius: 50%;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.c2::after {
  position: absolute;
  display: block;
  content: " ";
  width: 100%;
  height: 100%;
  background: linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.3), transparent);
  animation: 1.4s linear 0s infinite normal none running FDoGs;
  transform: translateX(-100%);
  inset: 0px;
}

.c1 {
  min-width: 100%;
  min-height: 100%;
  width: 100%;
  height: 100%;
  padding: initial;
  box-sizing: border-box;
  position: relative;
}

.c0 {
  min-width: 2.5rem;
  width: 2.5rem;
  max-width: 2.5rem;
  min-height: 2.5rem;
  height: 2.5rem;
  max-height: 2.5rem;
  vertical-align: super;
  border-radius: 50%;
  box-shadow: none;
  margin: 0.5rem;
  background: #FEFEFE;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
  </div>
</div>
`;

exports[`<AvatarImage /> should merge user classes & spread custom props to the root node 1`] = `
.c0 {
  min-width: 2.5rem;
  width: 2.5rem;
  max-width: 2.5rem;
  min-height: 2.5rem;
  height: 2.5rem;
  max-height: 2.5rem;
  vertical-align: super;
  border-radius: 50%;
  box-shadow: none;
  margin: 0.5rem;
  background: #FEFEFE;
}

.c1 {
  width: 100%;
  height: 100%;
  opacity: 1;
  object-fit: cover;
  border-radius: 50%;
  overflow: hidden;
}

<div
  className="c0 my-avatar"
  data-my-prop="woofAvatar"
>
  <img
    alt=""
    className="c1"
    referrerPolicy="no-referrer"
    src="https://storage.googleapis.com/enishi-common/images/user/person_default_icon.png"
  />
</div>
`;

exports[`<AvatarImage /> should merge user classes & spread custom props to the root node 2`] = `
.c0 {
  min-width: 2.5rem;
  width: 2.5rem;
  max-width: 2.5rem;
  min-height: 2.5rem;
  height: 2.5rem;
  max-height: 2.5rem;
  vertical-align: super;
  border-radius: 50%;
  box-shadow: none;
  margin: 0.5rem;
  background: #FEFEFE;
}

.c1 {
  width: 100%;
  height: 100%;
  opacity: 1;
  object-fit: cover;
  border-radius: 50%;
  overflow: hidden;
}

<div
  className="c0 my-avatar"
  data-my-prop="woofAvatar"
>
  <img
    alt=""
    className="c1"
    referrerPolicy="no-referrer"
    src="https://storage.googleapis.com/enishi-common/images/user/person_default_icon.png"
  />
</div>
`;

exports[`<AvatarImage /> should render as a button when button 1`] = `
.c0 {
  min-width: 2.5rem;
  width: 2.5rem;
  max-width: 2.5rem;
  min-height: 2.5rem;
  height: 2.5rem;
  max-height: 2.5rem;
  vertical-align: super;
  border-radius: 50%;
  box-shadow: none;
  margin: 0.5rem;
  background: #FEFEFE;
}

.c1 {
  cursor: pointer;
  position: relative;
  box-sizing: border-box;
  background: #FEFEFE;
  border-radius: 50%;
  transition: all 0.300s cubic-bezier(0.0, 0, 0.2, 1);
}

.c1:hover {
  box-shadow: 0px 0px 4px 2px rgb(0 0 0/16%);
}

.c1:hover>:nth-child(2) {
  transform: scale(1.2);
  animation: jkNVfK 0.6s ease-in-out 1.6s infinite alternate;
  background-color: #1F4682;
}

.c3 {
  background-color: transparent;
  position: absolute;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  inset: 0;
  transition: transform 0.300s cubic-bezier(0.0, 0, 0.2, 1),background-color 0.300s cubic-bezier(0.0, 0, 0.2, 1);
  transform: scale(1.1);
  animation: none;
  z-index: -1;
}

.c2 {
  width: 100%;
  height: 100%;
  opacity: 1;
  object-fit: cover;
  border-radius: 50%;
  overflow: hidden;
}

<div
  className="c0 c1"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  style={
    {
      "overflow": "hidden",
      "position": "relative",
      "verticalAlign": "top",
    }
  }
  tabIndex={0}
>
  <img
    alt=""
    className="c2"
    referrerPolicy="no-referrer"
    src="https://storage.googleapis.com/enishi-common/images/user/person_default_icon.png"
  />
  <div
    className="c3"
  />
</div>
`;
